src/hotspot/os/bsd/osThread_bsd.cpp

```diff
@@ -30,7 +30,7 @@
 
 OSThread::OSThread()
   : _thread_id(
-#ifdef __APPLE__
+#ifdef _ALLBSD_SOURCE
```

Comment:
This looks odd - does macOS define `_ALLBSD_SOURCE`?

Reply:
Yes, `_ALLBSD_SOURCE` includes macOS. See https://github.com/openjdk/jdk/pull/23611/files#diff-d0d775e9532fd48537bb0e77eceed944df53747cb59143375f92d74710d0ea22L455-L457

But I have to admit I don't really like that myself.

Reply:
Why is there even an ifdef for ALLBSD_SOURCE inside a file in the `os/bsd` directory? Afaik there are no instances where this could ever be undefined.

Reply:
@magicus That looks like my mistake :)

Reply:
Well, the original `#ifdef __APPLE__` was weird too, since prior to the upcoming BSD port, `bsd` just means "macos" in Hotspot code, so it does not make sense as it currently looks in the mainline either...